packages/trpc/server/routers/viewer/slots/util.ts

```diff
@@ -503,15 +503,75 @@ const _getAvailableSlots = async ({ input, ctx }: GetScheduleOptions): Promise<I
 
   let availableTimeSlots: typeof timeSlots = [];
   const bookerClientUid = ctx?.req?.cookies?.uid;
+  if (eventType.restrictionScheduleId) {
+    const restrictionSchedule = await prisma.schedule.findUnique({
+      where: { id: eventType.restrictionScheduleId },
+      select: {
+        id: true,
+        timeZone: true,
+        availability: {
+          select: {
+            days: true,
+            startTime: true,
+            endTime: true,
+            date: true,
+          },
+        },
+      },
+    });
+
+    if (restrictionSchedule) {
+      if (!eventType.useBookerTimezone && !restrictionSchedule.timeZone) {
```

Comment:
The code here from the first glance looks very similar to code in [handleNewBooking/ensureAvailableUsers.ts](https://github.com/calcom/cal.com/pull/21452/files#diff-cf280e896c971a9b613e58f3fca545dde37ff292d0326530cccc01df93e20ae9) - can we extract logic and re-use it?